### DISCUSSION TITLE
iOS Capacitor phone app cannot select audio output

### DISCUSSION THREAD

**Discussion by dchappelle** [ionic-team/capacitor#5401] [Q&A]
We have a web client that uses WebRTC to make audio and video calls. The web client uses the Quasar framework and we're now using Capacitor to run it as a hybrid app on Android and iOS. On both platforms, when a call is made, the audio is always routed to the device's speaker (unless there is e.g., a bluetooth headset attached). I created a plugin to toggle the speakerphone - which works on Android but not for iOS (15.2.1). I noticed for iOS, when looking at the audio session, there is only ever one output device available i.e., _iPhone speaker_ and trying to change the session category/mode doesn't work.

My question... is it possible to route the audio to other than the speaker for a Capacitor iOS phone app?

**Comment by Creiger** [comment]
Use cordova iosrtc plugin for ios webrtc connection and then you can also use it to change audio output between speaker and earpiece.

**Comment by angeloraso** [comment]
- I had the same problem and the cause was that the web view automatically changed the audio output device to speaker. The plugin worked fine but it was the web view that changed the output device
- btw, could you share your plugin?

**Reply by Creiger** [reply]
https://github.com/cordova-rtc/cordova-plugin-iosrtc
btw did you somehow manage to change audio output on iOS from speaker to earpiece and back?

**Reply by makamekm** [reply]
It does not work. It crashes on Android.

**Reply by Creiger** [reply]
Well it never supposed to work on android. it is called iOS RTC ;)
